### PR TITLE
to #7000: implement deleteTaskProcessRelation

### PR DESCRIPTION
**Pull request by zwZjut**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java [hunk 1]
```diff
@@ -191,15 +192,57 @@ private ProcessTaskRelationLog setRelationLog(ProcessDefinition processDefinitio
...
+        Map<String, Object> result = projectService.checkProjectAndAuth(loginUser, project, projectCode);
+        if (result.get(Constants.STATUS) != Status.SUCCESS) {
+            return result;
+        }
+        if (taskCode == 0) {
+            putMsg(result, Status.DELETE_TASK_PROCESS_RELATION_ERROR);
+            return result;
+        }
+        List<ProcessTaskRelation> downstreamList = processTaskRelationMapper.queryByCode(projectCode, processDefinitionCode, taskCode, 0L);
+        if (!CollectionUtils.isEmpty(downstreamList)) {
```
Comment (brave-lee) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java]: Maybe using `isNotEmpty` is more intuitive

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java [hunk 2]
```diff
@@ -191,15 +192,57 @@ private ProcessTaskRelationLog setRelationLog(ProcessDefinition processDefinitio
...
+            return result;
+        }
+        int deleteTaskDefinition = 0;
+        if (TaskType.CONDITIONS.getDesc().equals(taskDefinition.getTaskType())
+                || TaskType.DEPENDENT.getDesc().equals(taskDefinition.getTaskType())
+                || TaskType.SUB_PROCESS.getDesc().equals(taskDefinition.getTaskType())) {
+            deleteTaskDefinition = taskDefinitionMapper.deleteByCode(taskCode);
+        }
+
+        if (deleteRelation < 0 || deleteTaskDefinition < 0) {
```
Comment (brave-lee) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java]: Under what circumstances will less than ?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java [hunk 3]
```diff
@@ -191,15 +192,57 @@ private ProcessTaskRelationLog setRelationLog(ProcessDefinition processDefinitio
...
-     * @param projectCode project code
+     * @param loginUser             login user
+     * @param projectCode           project code
      * @param processDefinitionCode process definition code
-     * @param taskCode the post task code
+     * @param taskCode              the post task code
      * @return delete result code
      */
     @Override
+    @Transactional(rollbackFor = RuntimeException.class)
```
Comment (brave-lee) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java]: No exception thrown, transaction will not work.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java (hunk 1 above)
Comment (zwZjut) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java]: fixed

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java (hunk 2 above)
Comment (zwZjut) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java]: fixed

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java [hunk 4]
```diff
@@ -191,15 +192,57 @@ private ProcessTaskRelationLog setRelationLog(ProcessDefinition processDefinitio
...
+        if (null == taskDefinition) {
+            putMsg(result, Status.DATA_IS_NULL, "taskDefinition");
+            return result;
+        }
+        if (TaskType.CONDITIONS.getDesc().equals(taskDefinition.getTaskType())
+                || TaskType.DEPENDENT.getDesc().equals(taskDefinition.getTaskType())
+                || TaskType.SUB_PROCESS.getDesc().equals(taskDefinition.getTaskType())) {
+            int deleteTaskDefinition = taskDefinitionMapper.deleteByCode(taskCode);
+            if (0 == deleteTaskDefinition) {
+                putMsg(result, Status.DELETE_TASK_PROCESS_RELATION_ERROR);
```
Comment (JinyLeeChina) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java:242]: I think transaction control should be added here

**Review by brave-lee — APPROVED** [review]
+1

**Pull request merged**